src/main/java/com/minecolonies/coremod/network/messages/BuildToolPasteMessage.java

```diff
@@ -140,6 +174,62 @@ public void messageOnServerThread(final BuildToolPasteMessage message, final Ent
             StructureWrapper.loadAndPlaceStructureWithRotation(player.world, message.structureName,
                     message.pos, message.rotation, message.mirror ? Mirror.FRONT_BACK : Mirror.NONE, message.complete);
         }
+        else if(message.freeMode !=  null )
+        {
+            final List<ItemStack> stacks = new ArrayList<>();
+            final int chestHeight;
+
+            if(player.getStatFile().readStat(StatList.getObjectUseStats(ModItems.supplyChest)) > 0)
+            {
+                LanguageHandler.sendPlayerMessage(player, "com.minecolonies.coremod.error.supplyChestAlreadyPlaced");
+                return;
+            }
+
+            if(message.freeMode == WindowBuildTool.FreeMode.SUPPLYSHIP)
+            {
+                stacks.add(new ItemStack(ModItems.supplyChest));
+                chestHeight = 6;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 6 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS109)